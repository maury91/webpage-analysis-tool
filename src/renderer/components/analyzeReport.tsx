import * as React from 'react';
import { AnalyzePageResult } from '../../main/analyze';
import { Breadcrumbs, Button, Grid, Typography } from '@material-ui/core';
import { HorizontalBar } from 'react-chartjs-2';
import { transformDataToChart } from '../utils';
import { useState } from 'react';

interface AnalyzeDisplayProps {
  data: AnalyzePageResult;
}

export const AnalyzeReport = ({ data }: AnalyzeDisplayProps) => {
  const [showAll, setShowAll] = useState(false);
  const toggleShowAll = () => setShowAll(!showAll);
  const mostCommonTag = data.tags[0];
  return (
    <Grid container>
      <Grid item xs={12}>
        <HorizontalBar
          data={transformDataToChart(data, showAll ? Infinity : 5)}
          options={{
            title: {
              display: true,
              text: showAll ? 'All tags' : 'Top 5 tags'
            },
            scales: {
              xAxes: [
                {
                  ticks: {
                    min: 0
                  }
                }
              ]
            },
            legend: {
              display: false
            }
          }}
        />
      </Grid>
      <Grid item xs={12}>
        <Grid container justify="center">
          <Button variant="contained" color="secondary" onClick={toggleShowAll}>
            {showAll ? 'Show 5' : 'Show all'}
          </Button>
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <p>
          The most common used tag is <strong>{mostCommonTag.tag}</strong>, it appears{' '}
          <em>{mostCommonTag.count}</em> time{mostCommonTag.count > 1 ? 's' : ''}.
        </p>
        <p>
          The path where the tag is most used and deepest is:
          <p>
            <Breadcrumbs maxItems={5}>
              {data.mostUsedAndDeepestPath.split(' > ').map((path, index) => (
                <Typography key={index} color="textPrimary">
                  {path}
                </Typography>
              ))}
            </Breadcrumbs>
          </p>
        </p>
      </Grid>
    </Grid>
  );
};
