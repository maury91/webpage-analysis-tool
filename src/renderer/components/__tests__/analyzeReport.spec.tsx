import * as React from 'react';
import { shallow } from 'enzyme';
import { AnalyzeReport } from '../analyzeReport';

// Make the colors static instead of random
jest.mock('flat-colors', () => jest.fn().mockReturnValue([0, 0, 0]));

const analyzeData = {
  tags: [
    {
      tag: 'span',
      count: 3
    },
    {
      tag: 'div',
      count: 2
    },
    {
      tag: 'body',
      count: 1
    },
    {
      tag: 'html',
      count: 1
    }
  ],
  mostUsedAndDeepestPath: 'html > body > div > div > span'
};

describe('AnalyzeReport', () => {
  it('Simple snapshot testing', () => {
    const wrapper = shallow(<AnalyzeReport data={analyzeData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
