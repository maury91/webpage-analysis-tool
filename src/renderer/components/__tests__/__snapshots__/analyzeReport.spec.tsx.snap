// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyzeReport Simple snapshot testing 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <HorizontalBar
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "rgb(0,0,0)",
                "rgb(0,0,0)",
                "rgb(0,0,0)",
                "rgb(0,0,0)",
              ],
              "data": Array [
                3,
                2,
                1,
                1,
              ],
              "label": "",
            },
          ],
          "labels": Array [
            "span",
            "div",
            "body",
            "html",
          ],
        }
      }
      options={
        Object {
          "legend": Object {
            "display": false,
          },
          "scales": Object {
            "xAxes": Array [
              Object {
                "ticks": Object {
                  "min": 0,
                },
              },
            ],
          },
          "title": Object {
            "display": true,
            "text": "Top 5 tags",
          },
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Button))
        color="secondary"
        onClick={[Function]}
        variant="contained"
      >
        Show all
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <p>
      The most common used tag is 
      <strong>
        span
      </strong>
      , it appears
       
      <em>
        3
      </em>
       time
      s
      .
    </p>
    <p>
      The path where the tag is most used and deepest is:
      <p>
        <WithStyles(ForwardRef(Breadcrumbs))
          maxItems={5}
        >
          <WithStyles(ForwardRef(Typography))
            color="textPrimary"
            key="0"
          >
            html
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textPrimary"
            key="1"
          >
            body
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textPrimary"
            key="2"
          >
            div
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textPrimary"
            key="3"
          >
            div
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textPrimary"
            key="4"
          >
            span
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Breadcrumbs))>
      </p>
    </p>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
